SwiftyXMLParser/XML.swift

```diff
@@ -86,6 +86,13 @@ public func ?<< <T>(lhs: inout [T], rhs: T?) {
  ```
 */
 open class XML {
+
+    /**
+    Global XML setting for ignoring namespaces.
+    If set to true all accessors will ignore the first part of an element name up to a semicolon (:).
+     */
+    public static var ignoreNamespaces = false
```

Comment:
ignoreNamespaces should be a local property. Could you place it  on ```Element``` class?

```swift
extension XML {
    open class Element {
        open var name: String
        open var text: String?
        open var attributes = [String: String]()
        open var childElements = [Element]()
        open var lineNumberStart = -1
        open var lineNumberEnd = -1
        open var CDATA: Data?
        open var ignoreNamespaces: Bool // <-

        public init(name: String, ignoreNamespaces: Bool = false, text: String? = nil, attributes: [String: String] = [:], childElements: [Element] = []) {
```

and set the boolean value in ```Parser``` class

```swift
extension XML {
    class Parser: NSObject, XMLParserDelegate {
        // ....

        let ignoreNamespaces: Bool // <-

        // ...

        init(ignoreNamespaces: Bool = false) {
            self.ignoreNamespaces = ignoreNamespaces
            trimmingManner = nil
            documentRoot = Element(name: "XML.Parser.AbstructedDocumentRoot", ignoreNamespaces: ignoreNamespaces)
        }
        
        init(trimming manner: CharacterSet, ignoreNamespaces: Bool = false) {
            self.ignoreNamespaces = ignoreNamespaces
            trimmingManner = manner
            documentRoot = Element(name: "XML.Parser.AbstructedDocumentRoot", ignoreNamespaces: ignoreNamespaces)
        }
        
        // MARK:- private
        fileprivate var documentRoot: Element
        fileprivate var stack = [Element]()
        fileprivate let trimmingManner: CharacterSet?
        
        func parser(_ parser: XMLParser, didStartElement elementName: String, namespaceURI: String?, qualifiedName qName: String?, attributes attributeDict: [String : String]) {
            let node = Element(name: elementName, ignoreNamespaces: ignoreNamespaces) // <-
```

change ``` XML ``` I/F.

```swift
open class XML {

//    /**
//    Global XML setting for ignoring namespaces.
//    If set to true all accessors will ignore the first part of an element name up to a semicolon (:).
//     */
//    public static var ignoreNamespaces = false

    /**
    Interface to parse NSData
    
    - parameter data:NSData XML document
    - returns:Accessor object to access XML document
    */
    open class func parse(_ data: Data, ignoreNamespaces: Bool = false) -> Accessor {
        return Parser(ignoreNamespaces: ignoreNamespaces).parse(data)
    }
    
    /**
     Interface to parse String
     
     - Parameter str:String XML document
     - Returns:Accessor object to access XML document
     */
    open class func parse(_ str: String, ignoreNamespaces: Bool = false) throws -> Accessor { // <-
        guard let data = str.data(using: String.Encoding.utf8) else {
            throw XMLError.failToEncodeString
        }
        
        return Parser(ignoreNamespaces: ignoreNamespaces).parse(data)
    }
    
    /**
     Interface to parse NSData
     
     - parameter data:NSData XML document
     - parameter manner:NSCharacterSet If you wannna trim Text, assign this arg
     - returns:Accessor object to access XML document
     */
    open class func parse(_ data: Data, trimming manner: CharacterSet, ignoreNamespaces: Bool = false) -> Accessor { // <-
        return Parser(trimming: manner, ignoreNamespaces: ignoreNamespaces).parse(data)
    }
    
    /**
     Interface to parse String
     
     - Parameter str:String XML document
     - parameter manner:NSCharacterSet If you wannna trim Text, assign this arg
     - Returns:Accessor object to access XML document
     */
    open class func parse(_ str: String, trimming manner: CharacterSet, ignoreNamespaces: Bool = false) throws -> Accessor { // <-
        guard let data = str.data(using: String.Encoding.utf8) else {
            throw XMLError.failToEncodeString
        }
        
        return Parser(trimming: manner, ignoreNamespaces: ignoreNamespaces).parse(data)
    }
```